torch/csrc/autograd/python_cpp_function.h

```diff
@@ -22,7 +22,7 @@ PyObject* CppFunction_pynew(PyTypeObject *type, PyObject *args, PyObject *kwds)
   if (!obj) return NULL;
   THPCppFunction* f = (THPCppFunction*)obj.get();
   HANDLE_TH_ERRORS
-  new (&f->cdata) std::shared_ptr<Function>(Ctor()(args));
+  f->cdata.reset(Ctor()(args));
```

Comment:
I am not sure to understand why is the previous code a good thing to use (it is beyond my cpp knowledge), could you explain the difference?